java/src/main/java/com/logicalclocks/hsfs/engine/SparkEngine.java

```diff
@@ -95,6 +95,36 @@ private SparkEngine() {
     sparkSession.conf().set("spark.sql.hive.convertMetastoreParquet", "false");
   }
 
+  public void validateSparkConfiguration() throws FeatureStoreException {
+    String exceptionText = "Spark is misconfigured for communication with Hopsworks, missing or invalid property: ";
+
+    Map<String, String> configurationMap = new HashMap<>();
+    configurationMap.put("spark.hadoop.hops.ssl.trustore.name", null);
+    configurationMap.put("spark.hadoop.hops.rpc.socket.factory.class.default",
+            "io.hops.hadoop.shaded.org.apache.hadoop.net.HopsSSLSocketFactory");
+    configurationMap.put("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
+    configurationMap.put("spark.hadoop.hops.ssl.hostname.verifier", "ALLOW_ALL");
+    configurationMap.put("spark.hadoop.hops.ssl.keystore.name", null);
+    configurationMap.put("spark.hadoop.fs.hopsfs.impl", "io.hops.hopsfs.client.HopsFileSystem");
+    configurationMap.put("spark.hadoop.hops.ssl.keystores.passwd.name", null);
+    configurationMap.put("spark.hadoop.hops.ipc.server.ssl.enabled", "true");
+    configurationMap.put("spark.sql.hive.metastore.jars", null);
+    configurationMap.put("spark.hadoop.client.rpc.ssl.enabled.protocol", "TLSv1.2");
+    configurationMap.put("spark.hadoop.hive.metastore.uris", null);
+
+    for (Map.Entry<String, String> entry : configurationMap.entrySet()) {
+      if (entry.getValue() == null) {
+        if (sparkSession.conf().get(entry.getKey(), "").isEmpty()) {
+          throw new FeatureStoreException(exceptionText + entry.getKey());
+        }
+      } else {
+        if (!sparkSession.conf().get(entry.getKey(), "").equals(entry.getValue())) {
+          throw new FeatureStoreException(exceptionText + entry.getKey());
+        }
+      }
+    }
+  }
```

Comment:
Here the code can be simplified, also if the entry.value is null, you can use `contains` on the map, instead of doing the string comparison.

```suggestion
    for (Map.Entry<String, String> entry : configurationMap.entrySet()) {
      if (entry.getValue() == null && !sparkSession.conf().contains(entry.getKey())) {
        throw new FeatureStoreException(exceptionText + entry.getKey());
      } else if (!sparkSession.conf().get(entry.getKey(), "").equalsIgnoreCase(entry.getValue())) {
        throw new FeatureStoreException(exceptionText + entry.getKey());
      }
    }
```